packages/wallet/primitives/src/session-config.ts

```diff
@@ -342,6 +342,83 @@ export function encodeSessionsTopology(topology: SessionsTopology): Bytes.Bytes
   throw new Error('Invalid topology')
 }
 
+export function decodeSessionsTopology(bytes: Bytes.Bytes): SessionsTopology {
+  const { topology } = decodeSessionTopologyPointer(bytes)
+  return topology
+}
+
+function decodeSessionTopologyPointer(bytes: Bytes.Bytes): {
+  topology: SessionsTopology
+  pointer: number
+} {
+  if (bytes.length === 0) {
+    throw new Error('Empty topology bytes')
+  }
+
+  const flagByte = bytes[0]!
+  const flag = (flagByte & 0xf0) >> 4
+  const sizeSize = flagByte & 0x0f
+
+  if (flag === SESSIONS_FLAG_BRANCH) {
+    // Branch
+    if (sizeSize === 0 || sizeSize > 15) {
+      throw new Error('Invalid branch size')
+    }
+
+    let offset = 1
+    const encodedLength = Bytes.toNumber(bytes.slice(offset, offset + sizeSize))
+    offset += sizeSize
+
+    const encodedBranches = bytes.slice(offset, offset + encodedLength)
+    const branches: SessionsTopology[] = []
+
+    let branchOffset = 0
+    while (branchOffset < encodedBranches.length) {
+      const { topology: branchTopology, pointer: branchPointer } = decodeSessionTopologyPointer(
+        encodedBranches.slice(branchOffset),
+      )
+      branches.push(branchTopology)
+      branchOffset += branchPointer
+    }
+
+    return { topology: branches as SessionsTopology, pointer: offset + encodedLength }
+  } else if (flag === SESSIONS_FLAG_PERMISSIONS) {
+    // Permissions
+    const sessionPermissions = decodeSessionPermissions(bytes.slice(1))
+    const nodeLength = 1 + encodeSessionPermissions(sessionPermissions).length
+    return { topology: { type: 'session-permissions', ...sessionPermissions }, pointer: nodeLength }
+  } else if (flag === SESSIONS_FLAG_NODE) {
+    // Node
+    if (bytes.length < 33) {
```

Comment:
Magic number

Reply:
Thanks. I've added `SESSIONS_NODE_SIZE_BYTES`, this should always be a bytes32. The 33 includes the flag byte.